Compile error with latest version MFEM

Dear all,
Do you have any problem to build the latest version of MFEM in Windows 64bit + Visual Studio 2013? I used CMake to generate solution for VS2013 64bit version and build the solution, the following errors displayed:

Error	158	error C2259: 'mfem::ScalarProductInterpolator::AssembleElementMatrix2::ShapeCoefficient' : cannot instantiate abstract class	E:\mfem\mfem-master\fem\bilininteg.cpp	3075	1	mfem
Error	160	error C2259: 'mfem::VectorInnerProductInterpolator::AssembleElementMatrix2::VDotVShapeCoefficient' : cannot instantiate abstract class	E:\mfem\mfem-master\fem\bilininteg.cpp	3242	1	mfem
Error	157	error C2883: 'mfem::ScalarProductInterpolator::AssembleElementMatrix2::ShapeCoefficient::Eval' : function declaration conflicts with 'mfem::VectorCoefficient::Eval' introduced by using-declaration	E:\mfem\mfem-master\fem\bilininteg.cpp	3068	1	mfem
Error	159	error C2883: 'mfem::VectorInnerProductInterpolator::AssembleElementMatrix2::VDotVShapeCoefficient::Eval' : function declaration conflicts with 'mfem::VectorCoefficient::Eval' introduced by using-declaration	E:\mfem\mfem-master\fem\bilininteg.cpp	3234	1	mfem


Could anyone help me to take a look at it?
Thanks,
Tang Laoya

--- a/fem/bilininteg.cpp
+++ b/fem/bilininteg.cpp
@@ -3047,32 +3047,38 @@ void NormalInterpolator::AssembleElementMatrix2(
 }
 
 
+namespace internal
+{
+
+// Scalar shape functions scaled by scalar coefficient.
+// Used in the implementation of class ScalarProductInterpolator below.
+struct ShapeCoefficient : public VectorCoefficient
+{
+   Coefficient &Q;
+   const FiniteElement &fe;
+
+   ShapeCoefficient(Coefficient &q, const FiniteElement &fe_)
+      : VectorCoefficient(fe_.GetDof()), Q(q), fe(fe_) { }
+
+   using VectorCoefficient::Eval;
+   virtual void Eval(Vector &V, ElementTransformation &T,
+                     const IntegrationPoint &ip)
+   {
+      V.SetSize(vdim);
+      fe.CalcPhysShape(T, V);
+      V *= Q.Eval(T, ip);
+   }
+};
+
+}
+
 void
 ScalarProductInterpolator::AssembleElementMatrix2(const FiniteElement &dom_fe,
                                                   const FiniteElement &ran_fe,
                                                   ElementTransformation &Trans,
                                                   DenseMatrix &elmat)
 {
-   // Scalar shape functions scaled by scalar coefficient
-   struct ShapeCoefficient : public VectorCoefficient
-   {
-      Coefficient &Q;
-      const FiniteElement &fe;
-
-      ShapeCoefficient(Coefficient &q, const FiniteElement &fe_)
-         : VectorCoefficient(fe_.GetDof()), Q(q), fe(fe_) { }
-
-      using VectorCoefficient::Eval;
-      virtual void Eval(Vector &V, ElementTransformation &T,
-                        const IntegrationPoint &ip)
-      {
-         V.SetSize(vdim);
-         fe.CalcPhysShape(T, V);
-         V *= Q.Eval(T, ip);
-      }
-   };
-
-   ShapeCoefficient dom_shape_coeff(Q, dom_fe);
+   internal::ShapeCoefficient dom_shape_coeff(Q, dom_fe);
 
    elmat.SetSize(ran_fe.GetDof(),dom_fe.GetDof());
 
@@ -3209,6 +3215,35 @@ VectorCrossProductInterpolator::AssembleElementMatrix2(
 }
 
 
+namespace internal
+{
+
+// Vector shape functions dot product with a vector coefficient.
+// Used in the implementation of class VectorInnerProductInterpolator below.
+struct VDotVShapeCoefficient : public VectorCoefficient
+{
+   VectorCoefficient &VQ;
+   const FiniteElement &fe;
+   DenseMatrix vshape;
+   Vector vc;
+
+   VDotVShapeCoefficient(VectorCoefficient &vq, const FiniteElement &fe_)
+      : VectorCoefficient(fe_.GetDof()), VQ(vq), fe(fe_),
+        vshape(vdim, vq.GetVDim()), vc(vq.GetVDim()) { }
+
+   using VectorCoefficient::Eval;
+   virtual void Eval(Vector &V, ElementTransformation &T,
+                     const IntegrationPoint &ip)
+   {
+      V.SetSize(vdim);
+      VQ.Eval(vc, T, ip);
+      fe.CalcPhysVShape(T, vshape);
+      vshape.Mult(vc, V);
+   }
+};
+
+}
+
 void
 VectorInnerProductInterpolator::AssembleElementMatrix2(
    const FiniteElement &dom_fe,
@@ -3216,30 +3251,7 @@ VectorInnerProductInterpolator::AssembleElementMatrix2(
    ElementTransformation &Trans,
    DenseMatrix &elmat)
 {
-   // Vector shape functions dot product with a vector coefficient
-   struct VDotVShapeCoefficient : public VectorCoefficient
-   {
-      VectorCoefficient &VQ;
-      const FiniteElement &fe;
-      DenseMatrix vshape;
-      Vector vc;
-
-      VDotVShapeCoefficient(VectorCoefficient &vq, const FiniteElement &fe_)
-         : VectorCoefficient(fe_.GetDof()), VQ(vq), fe(fe_),
-           vshape(vdim, vq.GetVDim()), vc(vq.GetVDim()) { }
-
-      using VectorCoefficient::Eval;
-      virtual void Eval(Vector &V, ElementTransformation &T,
-                        const IntegrationPoint &ip)
-      {
-         V.SetSize(vdim);
-         VQ.Eval(vc, T, ip);
-         fe.CalcPhysVShape(T, vshape);
-         vshape.Mult(vc, V);
-      }
-   };
-
-   VDotVShapeCoefficient dom_shape_coeff(VQ, dom_fe);
+   internal::VDotVShapeCoefficient dom_shape_coeff(VQ, dom_fe);
 
    elmat.SetSize(ran_fe.GetDof(),dom_fe.GetDof());
 

